src/app/content/highlights/components/ColorPicker.tsx

```diff
@@ -104,22 +104,18 @@ const ColorPicker = ({className, ...props}: Props) => {
     },
     []
   );
-  const initialFocus = React.useCallback(
-    (event: React.FocusEvent) => {
-      if (!ref.current || event.currentTarget !== event.target) {
-        return;
-      }
-      const firstChecked = ref.current.querySelector<HTMLInputElement>('input[checked]');
-
-      if (firstChecked) {
-        firstChecked.focus();
-      } else {
-        ref.current.querySelector<HTMLInputElement>('input')?.focus();
+  const color = (props as SingleSelectProps).color;
+  const focusOnSelected = React.useCallback(
+    () => {
+      if (color && document?.activeElement === ref.current) {
+        ref.current?.querySelector<HTMLInputElement>(`input[name="${color}"]`)?.focus();
       }
     },
-    []
+    [color]
   );
 
+  React.useEffect(focusOnSelected, [focusOnSelected]);
```

Comment:
I think this will steal focus whenever this component is rendered... is that an OK thing?

Reply:
It will only change focus if the container is the active element (already has the focus - this is the 2nd part of the `if` in line 110).
The rule for radio groups is that the container is keyboard-focusable, and when it receives focus, the selected radio item gets focused (or if none is selected, the first one, but we always have a selection).

Reply:
ah ok I missed that